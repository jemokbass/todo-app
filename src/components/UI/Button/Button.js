import React from 'react';
import PropTypes from 'prop-types';

const Button = props => {
  const { type, children, onClick, disabled, className } = props;
  const buttonClassName = className ? ` ${className}` : '';

  return (
    <button className={`button${buttonClassName}`} type={type} onClick={onClick} disabled={disabled}>
      {children}
    </button>
  );
};

Button.defaultProps = {
  type: 'button',
  children: '',
  onClick: () => {},
  disabled: false,
};

Button.propTypes = {
  type: PropTypes.oneOf(['submit', 'button', 'reset']),
  children: PropTypes.node,
  onClick: PropTypes.func,
  className: PropTypes.string,
  disabled: PropTypes.bool,
};

export default Button;
