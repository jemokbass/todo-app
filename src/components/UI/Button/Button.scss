.button {
  display: block;
  width: 100%;
  padding: 10px 24px;
  font-weight: 700;
  font-size: 18px;
  color: $main-color;
  background: $element-color-lightest;
  border: $secondary-color;
  cursor: pointer;
  transition: all 0.5s ease 0s;

  &:hover {
    color: $text-color;
    background: $element-color-darkest;
  }
}

.button[disabled] {
  opacity: 0.6;
  cursor: not-allowed;

  &:hover {
    color: $main-color;
    background: $element-color-lightest;
  }
}
