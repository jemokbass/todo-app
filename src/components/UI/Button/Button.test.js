import React from 'react';
import { render, screen } from '@testing-library/react';

import Button from './Button';

describe('Button UI component', () => {
  it('Should render button component', () => {
    render(<Button />);
  });

  it('Should render button with text', () => {
    render(<Button>text</Button>);
    expect(screen.getByText(/text/i)).toBeInTheDocument();
  });

  describe('Should render snapshots', () => {
    it('Should render snapshot empty button', () => {
      const { asFragment } = render(<Button />);
      expect(asFragment(<Button />)).toMatchSnapshot();
    });

    it('Should render snapshot disabled button', () => {
      const { asFragment } = render(<Button disabled />);
      expect(asFragment(<Button disabled />)).toMatchSnapshot();
    });

    it('Should render snapshot button with text Win', () => {
      const { asFragment } = render(<Button>Win</Button>);
      expect(asFragment(<Button>Win</Button>)).toMatchSnapshot();
    });
  });

  describe('DefaultProps', () => {
    it('Should use default onClick', () => {
      const res = Button.defaultProps.onClick();
      expect(res).toBe(undefined);
    });

    it('Should use default disabled', () => {
      const res = Button.defaultProps.disabled;
      expect(res).toBe(false);
    });

    it('Should use default type button', () => {
      const res = Button.defaultProps.type;
      expect(res).toBe('button');
    });
  });
});
