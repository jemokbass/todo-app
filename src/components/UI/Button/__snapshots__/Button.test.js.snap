// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button UI component Should render snapshots Should render snapshot button with text Win 1`] = `
<DocumentFragment>
  <button
    class="button"
    type="button"
  >
    Win
  </button>
</DocumentFragment>
`;

exports[`Button UI component Should render snapshots Should render snapshot disabled button 1`] = `
<DocumentFragment>
  <button
    class="button"
    disabled=""
    type="button"
  />
</DocumentFragment>
`;

exports[`Button UI component Should render snapshots Should render snapshot empty button 1`] = `
<DocumentFragment>
  <button
    class="button"
    type="button"
  />
</DocumentFragment>
`;
